### PR TITLE
GH-1873 fixing incorrect cardinality calculation

### PR DESCRIPTION
**Pull request by reeshabhkumarranjan**
GitHub issue resolved: #1873 

* reassigned the `vars` list (list -> set -> list) so that it contains only distinct elements
* added a new package `optimizers` under `core/sail/federation/src/test/java/org/eclipse/rdf4j/sail/federation` as suggested by @hmottestad 
* added tests in a new file, `CardinalityTest.java` in the package `optimizers`, also added the header comment in the new file

### CODE REVIEW

File: core/sail/federation/src/test/java/org/eclipse/rdf4j/sail/federation/optimizers/CardinalityTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) ${year} Eclipse RDF4J contributors.
```
Comment (jeenbroekstra) [inline core/sail/federation/src/test/java/org/eclipse/rdf4j/sail/federation/optimizers/CardinalityTest.java]: Looks as if your editor forgot to replace `{$year}` with the actual year!

File: core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EvaluationStatistics.java [hunk 2]
```diff
@@ -82,6 +85,8 @@ public void meet(Slice slice) {
 
 		protected double getCardinality(StatementPattern pattern) {
 			List<Var> vars = pattern.getVarList();
+			Set<Var> varSet = new HashSet<Var>(vars);
+			vars = new ArrayList<Var>(varSet);
```
Comment (jeenbroekstra) [inline core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EvaluationStatistics.java]: I think you can also just do:

    Set<Var> vars = new HashSet<>(pattern.getVarList());

I don't think the rest of the method requires that `vars` is an ArrayList (or even a list), so no need to create an arraylist back out of the set. Saves having to copy the elements over twice.

File: core/sail/federation/src/test/java/org/eclipse/rdf4j/sail/federation/optimizers/CardinalityTest.java (hunk 1 above)
Comment (reeshabhkumarranjan) [reply core/sail/federation/src/test/java/org/eclipse/rdf4j/sail/federation/optimizers/CardinalityTest.java]: I saw other files, they have "2015" in place of `${year}`. Should I do the same or should I put "2020"?

File: core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EvaluationStatistics.java (hunk 2 above)
Comment (reeshabhkumarranjan) [reply core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EvaluationStatistics.java]: Alright! I will make this change and update this pull request.

File: core/sail/federation/src/test/java/org/eclipse/rdf4j/sail/federation/optimizers/CardinalityTest.java (hunk 1 above)
Comment (reeshabhkumarranjan) [reply core/sail/federation/src/test/java/org/eclipse/rdf4j/sail/federation/optimizers/CardinalityTest.java]: I replaced it with `2020` in the latest commit.

**Pull request merged**